### PR TITLE
Fix #33448: Format WP column "BCF snapshot" as string

### PR DESCRIPTION
**Pull request by wielinde**
Render an "X" if there is at least one viewpoint for the associated BCF issue.

https://community.openproject.org/wp/33448

### CODE REVIEW

File: app/models/work_package/exporter/formatters.rb [hunk 1]
```diff
@@ -1,19 +1,19 @@
 module WorkPackage::Exporter
   module Formatters
+    @@formatters = %i[default costs estimated_hours].map do |key|
```
Comment (oliverguenther) [inline app/models/work_package/exporter/formatters.rb]: I prefer the memoized style with an instance-level variable. Class variables are generally frowned upon due to the inheritance problems. Even though this module is likely not being subclassed, why introduce a class variable?

Comment (wielinde) [reply app/models/work_package/exporter/formatters.rb]: Yes, I already asked Jens how to improve that because I knew that it "smelled". I'll improve that. Thanks.

File: app/models/work_package/pdf_export/work_package_list_to_pdf.rb [hunk 2]
```diff
@@ -317,4 +317,10 @@ def batch_supported?
       false
     end
   end
+
+  def make_bcf_thumbnail_field_value(work_package)
```
Comment (oliverguenther) [inline app/models/work_package/pdf_export/work_package_list_to_pdf.rb]: This method is added but never used?

File: app/models/work_package/exporter/formatters.rb [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
+      Kernel.const_get("WorkPackage::Exporter::Formatters::#{key.to_s.camelize}")
+    end
+
     def self.all
-      @formatters ||= begin
-        %i[default costs estimated_hours].map do |key|
-          Kernel.const_get("WorkPackage::Exporter::Formatters::#{key.to_s.camelize}")
-        end
-      end
+      @@formatters
```
Comment (oliverguenther) [inline app/models/work_package/exporter/formatters.rb]: As stated above, I'd prefer to revert these changes into the instance variable pattern

**Review by oliverguenther — CHANGES_REQUESTED** [review]
Some minor code remarks

File: app/models/work_package/pdf_export/work_package_list_to_pdf.rb (hunk 2 above)
Comment (wielinde) [reply app/models/work_package/pdf_export/work_package_list_to_pdf.rb]: removed it.

File: app/models/work_package/exporter/formatters.rb (hunk 1 above)
Comment (wielinde) [reply app/models/work_package/exporter/formatters.rb]: changed it.

**Comment by wielinde** [comment]
**Pre review modification goals:**

- Make registering of formatters work. That is why I introduced the class variables.... bad idea.

**Post review modifications**
    
- Don't use class variables.
- Use memoized class instance variables.
- Fix class reloading in development environment.

**Comment by wielinde** [comment]
I don't see the point of the last remaining rubocop errors. `apply?` is a method of the code that gets tested. It is not code testing it.
